### PR TITLE
feat: [M3-6532] - Configuring main exports

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This configures our exports in the repo. Also added esLint so there's some extra diff because of that. I exported both the theme object and individuals tokens. This is all in flux and happy to see this evolve but I think this should get us a bit further. Added some comments for clarity

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -60,14 +68,47 @@ StyleDictionaryPackage.registerFilter({
...
+          } else {
+            removeMetadataAndFlatten(jsonObj[key]);
+          }
+        } else {
+          continue;
+        }
+      }
+      return jsonObj;
+    };
+    const flattenedTokens = removeMetadataAndFlatten(tokens);
```
Comment (abailly-akamai) [inline build.js:104]: the formatter provides a lot of metadata out of the box along with the token's value, which I believe we don't really need. This script will flatten them up and assign the `value` key's value to the token name. ex:

```JSON
"neutrals": {
      "grey07": {
        "value": "#555555",
        "type": "colors",
        "filePath": "tokens/globals/colors/colors.json",
        "isSource": true,
        "original": {
          "value": "{alias.colors.brand.neutrals.grey07}",
          "type": "colors"
        },
        "name": "TOKEN_COLOR_NEUTRALS_GREY07",
        "attributes": {},
        "path": [
          "colors",
          "neutrals",
          "grey07"
        ]
      },
}
```
becomes 
```JSON
"neutrals": {
      "grey07": "#555555",
}
```

Comment (abailly-akamai) [reply build.js:104]: I will comment and add tests if we're sticking with this

File: build.js [hunk 2]
```diff
@@ -60,14 +68,47 @@ StyleDictionaryPackage.registerFilter({
...
+
+    return `export default ${JSON.stringify(flattenedTokens, null, 2)
+      .replace(/"([^"]+)":/g, '$1:')
+      .replace('alias', 'aliases')
+      .replace('component', 'components')
+      .replace(/color/g, 'colors')}`;
+  },
```
Comment (abailly-akamai) [inline build.js:111]: cleanup job for outputing a sanitized string

replacing `alias`, `compoment` & `color` will need to be handled in a safer way

File: package.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
   "types": "build/index.d.ts",
   "private": false,
   "scripts": {
-    "build": "tsup src/index.ts --format esm --dts",
-    "generate": "node ./build.js",
+    "build": "tsup src/index.ts --format esm --dts --minify terser",
+    "generate": "yarn clean && node ./build.js && yarn build",
```
Comment (abailly-akamai) [inline package.json:9]: yarn `generate` to build the whole thing

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const akamaiTheme: NodeRequire = require('../dist/web/akamai/theme.es6.js');
+const akamaiTokens: NodeRequire = require('../dist/web/akamai/tokens.es6.js');
+const cloudManagerTheme: NodeRequire = require('../dist/web/cloudmanager/theme.es6.js');
+const cloudManagerTokens: NodeRequire = require('../dist/web/cloudmanager/tokens.es6.js');
+
+export { akamaiTheme, akamaiTokens, cloudManagerTheme, cloudManagerTokens };
```
Comment (abailly-akamai) [inline src/index.ts:6]: Not tied to naming conventions here. I also don't know if we actually want to export individual tokens, but I also can see it being useful for one-offs outside of the theme definition

**Comment by jaalah-akamai** [comment]
This looks great! Nice work 🚀

File: build.js [hunk 5]
```diff
@@ -17,9 +20,14 @@ function getStyleDictionaryConfig(brand, platform) {
...
-          }
-        ]
+            filter: 'excludeBrand',
+          },
+          {
+            destination: 'theme.es6.js',
+            format: 'javascript/nested',
+            filter: 'excludeBrand',
+          },
+        ],
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply build.js:29]: Good idea - done!

File: src/index.ts (hunk 4 above)
Comment (abailly-akamai) [reply src/index.ts:6]: this one is giving me quite a bit of problems so I'll leave it for M3-6533

**Comment by abailly-akamai** [comment]
@jaalah-akamai addressed comments minus one that we can handle as a TS enhancement in M3-6533

Feel free to merge once approved!

**Review by jaalah-akamai — APPROVED** [review]

**Pull request merged**